src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -3,20 +3,45 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.machine.Machine;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
 public class MachineDaoImpl extends AbstractDao implements MachineDao {
-    protected MachineDaoImpl(SessionFactory sessionFactory) {
+    public MachineDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add machine to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Machine> query = session.createQuery("from Machine "
+                    + "where year > :age", Machine.class);
+            query.setParameter("age", age);
```

Comment:
technically `year` column represents manifacturing year, and `age` - the age of the machine in years